airgun/widgets.py

```diff
@@ -1267,6 +1267,89 @@ def read(self):
             "return ace.edit('{0}').getValue();".format(self.ace_edit_id))
 
 
+class Pagination(Widget):
+    """Represents Paginator widget that includes per page selector, First/Last/Next/Prev buttons
+    and current page index/overall amount of pages. Mainly used with Table widget.
+    """
+    ROOT = "//form[contains(@class, 'content-view-pf-pagination')]"
+    # Kattelo views use per_page with select, foreman use a per_page with Button DropDown.
+    PER_PAGE_BUTTON_DROPDOWN = ".//div[button[@id='pagination-row-dropdown']]"
+    PER_PAGE_SELECT = ".//select[contains(@ng-model, 'per_page')]"
+    first_page_button = Text(".//li[a[span[contains(@class, 'angle-double-left')]]]")
+    previous_page_button = Text(".//li[a[span[contains(@class, 'angle-left')]]]")
+    next_page_button = Text(".//li[a[span[contains(@class, 'angle-right')]]]")
+    last_page_button = Text(".//li[a[span[contains(@class, 'angle-double-right')]]]")
+    page = TextInput(locator=".//input[contains(@class, 'pagination-pf-page')]")
+    pages = Text(".//span[contains(@class, 'pagination-pf-pages')]")
+
+    def _element_exists(self, locator, parent=None, check_visibility=False):
+        """Check if a locator element/widget exists in the DOM"""
+        try:
```

Comment:
why not?
```
        try:
            self.browser.elements(
                locator, parent=parent, check_visibility=check_visibility))
        except NoSuchElementException:
            return False
        return True
```

Reply:
do not want to use hidden intention (I understand that if no exception => we have found visible elements), The code supplied really translate our intentions of what we want.

Reply:
or maybe `return self.browser.wait_for_element( ... timeout=0.1, exception=False) is not None` if you really want to make this method shorter. But i'm ok with any option.

Reply:
@abalakh we do want to wait for that element, it's here visible or not.